From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander Khabarov <alexander.khabarov@arm.com>
Date: Thu, 19 Oct 2023 15:00:17 +0000
Subject: [PATCH] Add Bazel LLVM targets

The repository configuration step of the overlay generates the
`llvm/targets.bzl`. We can generate it ourselves by using the
`DEFAULT_TARGETS`:

```sh
tr '\n' ' ' < utils/bazel/configure.bzl |
grep -o 'DEFAULT_TARGETS *= *\[[^]]\+\]' |
sed \
  -e 's|DEFAULT_TARGETS|llvm_targets|' \
  -e 's|,\( *\)|,\n\1|g' \
  -e 's|\[|[\n|' \
> llvm/targets.bzl
```
---
 llvm/targets.bzl | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)
 create mode 100644 llvm/targets.bzl

diff --git a/llvm/targets.bzl b/llvm/targets.bzl
new file mode 100644
index 000000000..7a31d801e
--- /dev/null
+++ b/llvm/targets.bzl
@@ -0,0 +1,21 @@
+llvm_targets = [
+    "AArch64",
+    "AMDGPU",
+    "ARM",
+    "AVR",
+    "BPF",
+    "Hexagon",
+    "Lanai",
+    "LoongArch",
+    "Mips",
+    "MSP430",
+    "NVPTX",
+    "PowerPC",
+    "RISCV",
+    "Sparc",
+    "SystemZ",
+    "VE",
+    "WebAssembly",
+    "X86",
+    "XCore",
+]
